### PR TITLE
Bump to clang-13 to get some C++20 support

### PR DESCRIPTION
**Pull request by nilsdeppe**
## Proposed changes

Switching to C++20 will be done in a separate PR.

### Upgrade instructions

We don't guarantee compatibility with Clang < 13 anymore. Upgrade to Clang 13+ or use GCC 9+.

### Code review checklist

- [ ] The code is documented and the documentation renders correctly. Run
  `make doc` to generate the documentation locally into `BUILD_DIR/docs/html`.
  Then open `index.html`.
- [ ] The code follows the stylistic and code quality guidelines listed in the
  [code review guide](https://spectre-code.org/code_review_guide.html).
- [ ] The PR lists upgrade instructions and is labeled `bugfix` or
  `new feature` if appropriate.

### CODE REVIEW

File: docs/Installation/Installation.md [hunk 1]
```diff
@@ -58,7 +58,9 @@ all of these dependencies.
 #### Required:
 
 * [GCC](https://gcc.gnu.org/) 9.1 or later,
-[Clang](https://clang.llvm.org/) 11.0 or later, or AppleClang 11.0.0 or later
+[Clang](https://clang.llvm.org/) 12.0 or later (see
```
Comment (nilsvu) [inline docs/Installation/Installation.md]: There's no test for clang 12. Do you want to add one or bump to clang 13?

File: docs/Installation/Installation.md [hunk 2]
```diff
@@ -58,7 +58,9 @@ all of these dependencies.
...
+[here](https://apt.llvm.org/) for how to get newer versions of clang through
+apt), or AppleClang 11.0.0 or later
```
Comment (nilsvu) [inline docs/Installation/Installation.md]: You can bump AppleClang to 13 as well, since that's the version we're testing on CI.

**Review by nilsvu — CHANGES_REQUESTED** [review]

File: docs/Installation/Installation.md (hunk 1 above)
Comment (nilsdeppe) [reply docs/Installation/Installation.md]: We can bump to 13, I think. Wheeler has clang-13 installed.

**Comment by nilsdeppe** [comment]
Okay, bumped to clang-13 (I think that's what we'd agreed on is some discussions anyway)

File: docs/Installation/Installation.md (hunk 1 above)
Comment (knelli2) [reply docs/Installation/Installation.md]: Ocean also has llvm 13 installed so all good there

**Review by knelli2 — APPROVED** [review]

File: docs/Installation/Installation.md (hunk 1 above)
Comment (nilsvu) [reply docs/Installation/Installation.md]: CaltechHPC has only clang 12, but I don't care

Comment (nilsdeppe) [reply docs/Installation/Installation.md]: Sounds good. And we can ask them to update. That should be doable.

Comment (nilsvu) [reply docs/Installation/Installation.md]: I'll open a ticket with them.

**Review by nilsvu — APPROVED** [review]

**Review by wthrowe — APPROVED** [review]

**Review by kidder — APPROVED** [review]

**Pull request merged**